source/modulo_components/src/LifecycleComponent.cpp

```diff
@@ -7,74 +7,99 @@ namespace modulo_components {
+          );
+          interface =
+              std::make_shared<PublisherHandler<rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::Float64MultiArray>,
+                                                std_msgs::msg::Float64MultiArray>>(
+                  PublisherType::LIFECYCLE_PUBLISHER, publisher
+              )->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::INT32: {
+          auto publisher = this->create_publisher<std_msgs::msg::Int32>(topic_name, this->qos_);
+          interface = std::make_shared<PublisherHandler<rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::Int32>,
+                                                        std_msgs::msg::Int32>>(
+              PublisherType::LIFECYCLE_PUBLISHER, publisher
+          )->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::STRING: {
+          auto publisher = this->create_publisher<std_msgs::msg::String>(topic_name, this->qos_);
+          interface = std::make_shared<PublisherHandler<rclcpp_lifecycle::LifecyclePublisher<std_msgs::msg::String>,
+                                                        std_msgs::msg::String>>(
+              PublisherType::LIFECYCLE_PUBLISHER, publisher
+          )->create_publisher_interface(message_pair);
+          break;
+        }
+        case MessageType::ENCODED_STATE: {
+          auto publisher = this->create_publisher<modulo_new_core::EncodedState>(topic_name, this->qos_);
+          interface =
+              std::make_shared<PublisherHandler<rclcpp_lifecycle::LifecyclePublisher<modulo_new_core::EncodedState>,
+                                                modulo_new_core::EncodedState>>(
+                  PublisherType::LIFECYCLE_PUBLISHER, publisher
+              )->create_publisher_interface(message_pair);
+          break;
+        }
       }
+    } catch (const std::exception& ex) {
+      success = false;
+      RCLCPP_ERROR_STREAM(this->get_logger(), "Failed to configure output '" << name << "': " << ex.what());
     }
   }
+  return success;
 }
 
-void LifecycleComponent::activate_outputs() {
-  // FIXME need to catch exceptions?
+bool LifecycleComponent::activate_outputs() {
+  bool success = true;
   for (auto const& [name, interface]: this->outputs_) {
-    interface->activate();
+    try {
+      interface->activate();
+    } catch (const std::exception& ex) {
+      success = false;
+      RCLCPP_ERROR_STREAM(this->get_logger(), "Failed to activate output '" << name << "': " << ex.what());
+    }
+  }
+  return success;
+}
+
+bool LifecycleComponent::deactivate_outputs() {
+  bool success = true;
+  for (auto const& [name, interface]: this->outputs_) {
+    try {
+      interface->deactivate();
+    } catch (const std::exception& ex) {
+      success = false;
+      RCLCPP_ERROR_STREAM(this->get_logger(), "Failed to deactivate output '" << name << "': " << ex.what());
+    }
   }
+  return success;
```

Comment:
I think you can avoid the intermediate variable in these functions and just return, since it's happening at the end of the function.
```suggestion
      RCLCPP_ERROR_STREAM(this->get_logger(), "Failed to deactivate output '" << name << "': " << ex.what());
      return false;
    }
  }
  return true;
```

Reply:
But it's in for loops so I was thinking we should try to activate/deactivate as many outputs as possible and not return as soon as one output fails. Not good?

Reply:
Oh yes of course, I missed the alignment of the braces because of the diff formatting, indeed that's better like you have it